contracts/service-provider-directory/src/state/services.rs

```diff
@@ -55,39 +55,38 @@ pub fn has_service(store: &dyn Storage, service_id: ServiceId) -> bool {
 
 pub fn load_id(store: &dyn Storage, service_id: ServiceId) -> Result<Service> {
     services().load(store, service_id).map_err(|err| match err {
-        StdError::NotFound { .. } => ContractError::NotFound { service_id },
+        StdError::NotFound { .. } => SpContractError::NotFound { service_id },
         err => err.into(),
     })
 }
 
-pub fn load_announcer(store: &dyn Storage, announcer: Addr) -> Result<Vec<(ServiceId, Service)>> {
+pub fn load_announcer(store: &dyn Storage, announcer: Addr) -> Result<Vec<Service>> {
     let services = services()
         .idx
         .announcer
         .prefix(announcer)
         .range(store, None, None, Order::Ascending)
         .take(MAX_NUMBER_OF_PROVIDERS_PER_ANNOUNCER as usize)
+        .map(|res| res.map(|item| item.1))
         .collect::<StdResult<Vec<_>>>()?;
     Ok(services)
 }
 
-pub fn load_nym_address(
-    store: &dyn Storage,
-    nym_address: NymAddress,
-) -> Result<Vec<(ServiceId, Service)>> {
+pub fn load_nym_address(store: &dyn Storage, nym_address: NymAddress) -> Result<Vec<Service>> {
     let services = services()
         .idx
         .nym_address
         .prefix(nym_address.to_string())
         .range(store, None, None, Order::Ascending)
         .take(MAX_NUMBER_OF_ALIASES_FOR_NYM_ADDRESS as usize)
```

Comment:
given that clients shouldn't be allowed to store more than `MAX_NUMBER_OF_ALIASES_FOR_NYM_ADDRESS` values, do we even need that `take` here?

Reply:
In theory not, but defensive coding